subsys/net/lib/prometheus/Kconfig

```diff
@@ -0,0 +1,31 @@
+# Copyright(c) 2024 Sparse Technology, Mustafa Abdullah Kus
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig PROMETHEUS
+	bool "Prometheus Client Library (Pull Method)"
+	depends on NET_SOCKETS
+	depends on HTTP_SERVER
+	help
+	  Enable Prometheus client library
+
+if PROMETHEUS
+
+module = PROMETHEUS
+module-dep = NET_LOG
+module-str = Log level for PROMETHEUS
+module-help = Enable debug message of PROMETHEUS client library.
+source "subsys/net/Kconfig.template.log_config.net"
+
+config PROMETHEUS_MAX_METRICS
+	int "Maximum number of metrics"
+	default 10
+	help
+	  Maximum number of metrics that can be registered.
+
+config PROMETHEUS_METRIC_FORMAT_BUFFER_SIZE
+	int "Exposition buffer size"
+	default 256
+	help
+	  Size of the buffer that holds the exposition data.
```

Comment:
See above if interested in a small optimization which permits to remove this temporary buffer.

Reply:
Removed with updated related `write_to_buffer` parts.